### DISCUSSION TITLE
How to integrate with clang-tidy?

### DISCUSSION THREAD

**Discussion by jeremy-coulon** [ccache/ccache#1052] [Questions & answers]
Hi,

I would like to know if anybody has tried to integrate ccache and clang-tidy together.

Can I write a wrapper script that could do what I want?

Here is how I currently launch clang-tidy:
```
$ run-clang-tidy -export-fixes clang-tidy.yaml -checks=-*,modernize-use-nullptr
Enabled checks:
    modernize-use-nullptr

clang-apply-replacements version 10.0.0
clang-tidy-10 -checks=-*,modernize-use-nullptr -export-fixes /tmp/tmpxgnsdyg4/tmpr41frdgx.yaml -p=/data/myproject/build /data/myproject/src/file1.cpp
clang-tidy-10 -checks=-*,modernize-use-nullptr -export-fixes /tmp/tmpxgnsdyg4/tmp92332fg4.yaml -p=/data/myproject/build /data/myproject/src/file2.cpp
[...]
clang-tidy-10 -checks=-*,modernize-use-nullptr -export-fixes /tmp/tmpxgnsdyg4/tmp92399fg4.yaml -p=/data/myproject/build /data/myproject/src/file42.cpp
Writing fixes to clang-tidy.yaml ...
```

`-export-fixes /tmp/tmpxgnsdyg4/tmp92332fg4.yaml` is a temporary file with a generated name that changes for each run. This is the result file that I would need to cache.

`run-clang-tidy` can be configured with `-clang-tidy-binary mywrapper.py` in order to execute `mywrapper.py` instead of `clang-tidy-10` for each of my source file.

I don't know how to call ccache in my wrapper in order to make it work.
I have seen some ccache options in order to control which arguments should be considered for computing hash key.
How can I tell ccache what output file should be cached (the result of the compilation)?

**Comment by mgehre-amd** [comment]
I think the main change would be teaching ccache about where to find the corresponding .clang-tidy file and to cache it.
Beside that, ccache would need to learn to identify clang-tidy and understand how to hash the command line arguments.
